### PR TITLE
Add support to "dump", "populate" and "cleanup" parameters

### PR DESCRIPTION
**Pull request by henrydekap**

### CODE REVIEW

**Comment by codemedic** [comment]
@henrydekap sorry about the late response; been busy lately.
I will review it as soon as I can.

**Comment by henrydekap** [comment]
don't worry @codemedic, I don't have deadlines for this! :smiley:

**Comment by rbellido54** [comment]
Nice! Is there work needed to be done for the 2.0 branch as well?

**Comment by eventbase-web** [comment]
I'm eagerly awaiting this PR being merged. Anything I can do to help speed it along?

File: src/Codeception/Extension/MultiDb.php [hunk 1]
```diff
@@ -85,6 +86,55 @@ public function _initialize()
...
+            if ($connectorConfig['populate']) {
+                if ($connectorConfig['cleanup']) {
+                    $this->cleanup($connector);
+                }
+                $this->loadDump($connector);
+            }
+        }
+    }
+
+    protected function loadDump($connector)
```
Comment (codemedic) [inline src/Codeception/Extension/MultiDb.php:100]: PHPDoc Comments missing; but never mind I will add that this time :)

File: src/Codeception/Extension/MultiDb.php [hunk 2]
```diff
@@ -85,6 +86,55 @@ public function _initialize()
...
+            $this->getDriver($connector)->load($sql);
+        } catch (\PDOException $e) {
+            throw new ModuleException(
+              __CLASS__,
+              $e->getMessage() . "\nSQL query being executed: " . $sql
+            );
+        }
+    }
+
+    protected function cleanup($connector)
```
Comment (codemedic) [inline src/Codeception/Extension/MultiDb.php:131]: PHPDoc Comments missing; but never mind I will add that this time :)

File: src/Codeception/Extension/MultiDb.php [hunk 3]
```diff
@@ -13,6 +13,7 @@
 use Codeception\Exception\ModuleConfig as ModuleConfigException;
 use Codeception\Module;
 use Codeception\TestCase;
+use Codeception\Configuration as Configuration;
```
Comment (codemedic) [inline src/Codeception/Extension/MultiDb.php:16]: Unnecessary aliasing; really dont need the `as Configuration` part.

File: src/Codeception/Extension/MultiDb.php [hunk 4]
```diff
@@ -85,6 +86,55 @@ public function _initialize()
...
+                    Please, check path for sql file: " . $config['dump']
+                );
+            }
+            $sql = file_get_contents(Configuration::projectDir() . $config['dump']);
+            $sql = preg_replace('%/\*(?!!\d+)(?:(?!\*/).)*\*/%s', "", $sql);
+            if (!empty($sql)) {
+                $sql = explode("\n", $sql);
+            }
+        }
+        if (!$sql) {
```
Comment (codemedic) [inline src/Codeception/Extension/MultiDb.php:118]: This `if` and the `try/catch` below (with its contents) can be merged with `if (!empty($sql)) {` block.

**Comment by codemedic** [comment]
I have made some comments for the sake of clarity; I will make those changes and merge.
Nothing for you to act on @henrydekap

**Pull request merged**